src/model/maths/vec2.rs

```diff
@@ -1,4 +1,146 @@
+use std::ops::{Add, Mul, Sub};
+use std::cmp::PartialEq;
+use std::fmt::{Display, Formatter, Result};
+
+#[derive(Debug)]
 pub struct Vec2 {
     x: f64,
     y: f64
+}
+
+impl Vec2 {
+	pub fn new(x: f64, y: f64) -> Self {
+		Self { x, y }
+	}
+
+	pub fn x(&self) -> &f64 {
+		&self.x
+	}
+
+	pub fn y(&self) -> &f64 {
+		&self.y
+	}
+
+	pub fn dot(&self, other: &Self) -> f64 {
+		self.x * other.x + self.y * other.y
+	}
+
+	pub fn get_norm(&self) -> f64 {
+		(self.x * self.x + self.y * self.y).sqrt()
+	}
+
+	pub fn normalize(&self) -> Self {
+		let norm: f64 = self.get_norm();
+		Self {
+			x: self.x / norm,
+			y: self.y / norm
+		}
+	}
+}
+
+impl Add for Vec2 {
+	type Output = Self;
+	fn add(self, rhs: Self) -> Self::Output {
+		Self {
+			x: self.x + &rhs.x,
+			y: self.y + &rhs.y
+		}
+	}
+}
+
+impl Sub for Vec2 {
+	type Output = Self;
+	fn sub(self, rhs: Self) -> Self::Output {
+		Self {
+			x: self.x - &rhs.x,
+			y: self.y - &rhs.y
+		}
+	}
+}
+
+impl Mul<Self> for Vec2 {
+	type Output = Self;
+	fn mul(self, rhs: Self) -> Self::Output {
+		Self {
+			x: self.x * rhs.x,
+			y: self.y * rhs.y
+		}
+	}
+}
+
+impl Mul<f64> for Vec2 {
+	type Output = Self;
+	fn mul(self, rhs: f64) -> Self::Output {
+		Self {
+			x: self.x * rhs,
+			y: self.y * rhs
+		}
+	}
+}
+
+impl PartialEq for Vec2 {
```

Comment:
Tu peux simplement annoter #[derive(PartielEq)] sur ta struct, il va comparer automatiquement les membres de ta struct si eux memes implementent PartielEq (ce qui est le cas des types primaires)

Reply:
Et tant que tu es derive(Eq) aussi, qui est une une equivalence stricte (commutative ou non)

Reply:
the trait bound `f64: Eq` is not satisfied
the following other types implement trait `Eq`:
  i128
  i16
  i32
  i64
  i8
  isize
  u128
  u16
and 4 others